src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -341,7 +315,12 @@ private Stream TryConnectToProcess(int nodeProcessId, int timeout, long hostHand
             // Try and connect to the process.
             string pipeName = "MSBuild" + nodeProcessId;
 
-            NamedPipeClientStream nodeStream = new NamedPipeClientStream(".", pipeName, PipeDirection.InOut, PipeOptions.Asynchronous);
+            PipeOptions pipeOptions = PipeOptions.Asynchronous;
+
+            // TODO: when https://github.com/dotnet/corefx/issues/25427 is available
+            // pipeOptions |= PipeOptions.CurrentUserOnly;
```

Comment:
We need this before we can deploy this option, or we'll have regressed the pipe security guards on .NET Core/Windows.